### PR TITLE
ADR for restricting access to content by organisation

### PR DESCRIPTION
**Pull request by mtaylorgds**
An architecture decision record to record the approach to be used for restricting access to publications in Mainstream Publisher.

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: docs/arch/adr-011-restricting-access-to-content-by-organisation.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+- The Primary Publishing Organisation field is an existing field, and so this solution does not require introducing any new fields/concepts (though it is potentially changing the meaning of the association slightly by using it for access control).
+
+### Questions
+
+The Primary Publishing Organisation data flows through the Publishing API and into the content stored in the Content Store. Does it make sense to be changing this data to support access control within Mainstream Publisher (it introduces a level of coupling between two related, but not identical, concepts).
+
+### Option 2: Introducing a new field on content items
+
+A new field could be introduced to content items within Mainstream Publisher to store the organisation(s) whose members should have access to that content item.
```
Comment (baisa) [inline docs/arch/adr-011-restricting-access-to-content-by-organisation.md:47]: Maybe we could add a note to refer to [slack thread](https://gds.slack.com/archives/C051S690LGL/p1720451970243779) pointing that even though this solution may be confusing for future devs, hardcoded PPO is currently not exposed to users, so they would not be aware of this potential data duplication.

File: docs/arch/adr-011-restricting-access-to-content-by-organisation.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+This ADS focuses on the technical solution to restrict access, using the data supplied by Signon about the user (i.e. the organisation to which they belong).
+
+### Option 1: Use the primary publishing organisation
+
+All content that is published on GOV.UK by Mainstream Publisher has a `primary_publishing_organisation` association set on it. This is currently always set to GDS at the point of publishing the content, but we could store this association as part of the content item in Mainstream Publisher, and set it to the relevant organisation. When determining whether a user should have access to a content item, it's Primary Publishing Organisation field would be compared with the user's organisation.
+
+#### Disadvantages
+
+- The Primary Publishing Organisation association already exists for a purpose, and is not intended as a means of access control.
```
[the remark this answers is not in the document]
Comment (mtaylorgds) [reply docs/arch/adr-011-restricting-access-to-content-by-organisation.md:33]: I'm not really sure what this association's purpose is!

According to the [publishing API content schema](https://github.com/alphagov/publishing-api/blob/3c5de5f4c474c640abbc16ca66381be4234f218a/content_schemas/formats/shared/base_links.jsonnet#L14) it's "The organisation that published the page" (it also goes on to say that it's only set by Whitehall, which is clearly not true!), but why it exists, and what other parts of the system might be using it, I don't know.

If you have more knowledge about it, then I'm happy to update based on that.

File: docs/arch/adr-011-restricting-access-to-content-by-organisation.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+## Decision
+
+We will implement [option 2](#option-2-introducing-a-new-field-on-content-items). This keeps the field focused on the purpose of authorisation, and also limits the impact to be purely within Mainstream Publisher.
+
+GDS content designers (i.e. those with the "Editor" permission in Signon) will continue to have full access to all Content Items in Mainstream Publisher.
+
+## Consequences
+
+The data for this new field will need to be populated somehow. This will likely be done by creating a rake task to populate the field. The actual data to be used will need to be sourced, for example, by using the existing tagging association to organisations.
```
[the remark this answers is not in the document]
Comment (mtaylorgds) [reply docs/arch/adr-011-restricting-access-to-content-by-organisation.md]: I've added some text to explain this.

File: docs/arch/adr-011-restricting-access-to-content-by-organisation.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+There is a wider piece of work looking at access and permissions across the Publishing estate, but access to Mainstream Publisher needs to be widened before that work will complete.
+
+Therefore, we need an interim solution that restricts access within Mainstream Publisher to a subset of the content held within it, acknowledging that how this works might change longer-term.
+
+In the immediate term, the requirement is to limit access, for non-GDS content designers, to that content which is managed by their own department. Signon does provide the organisation that a user belongs to, which Mainstream Publisher will use in order to make authorisation decisions when attempting to view and edit content. There are some cases where a content item would need to be managed by content designers from multiple organisations. 
+
+For now the access rules can be binary, with users either given full access to a piece of content (view/edit/publish), or no access.
```
[the remark this answers is not in the document]
Comment (mtaylorgds) [reply docs/arch/adr-011-restricting-access-to-content-by-organisation.md]: I've added some text to address this.

File: docs/arch/adr-011-restricting-access-to-content-by-organisation.md (hunk 1 above)
Comment (mtaylorgds) [reply docs/arch/adr-011-restricting-access-to-content-by-organisation.md:49]: I've added some text.

**Review by baisa — APPROVED** [review]
Great work!

**Pull request merged**